sys/include/ztimer.h

```diff
@@ -465,6 +465,20 @@ static inline ztimer_now_t ztimer_now(ztimer_clock_t *clock)
     }
 }
 
+/**
+ * @brief   Get the time until a timer will trigger
+ *
+ *          O(n): iterate over all timers active on the clock until timer is reached.
+ *          n: being the timers active on the clock that are earlier to trigger
+ *             than the timer this is measuring to
+ *
```

Comment:
```suggestion
 *
 * @pre  The timer has been set on a clock.
 *
```

That's not a requirement of the code (it works fine with an uninitialized timer), but a) we don't need to allow that (especially because uninitialized means it could take the slow path), and b) it helps keep the return value definition concise (which'd otherwise need to say "or has already triggered or was never set")